api/samples/CMakeLists.txt

```diff
@@ -314,9 +312,7 @@ DR_install(FILES "${public_clist}"
   DESTINATION "${INSTALL_SAMPLES}" RENAME CMakeLists.txt)
 if (WIN32)                                                          # NON-PUBLIC
   # Have a copy of dynamorio library for tracedump.                 # NON-PUBLIC
-  # XXX i#1557: replace w/ generator expression to satisfy CMP0026. # NON-PUBLIC
-  get_target_property(DR_TARGET_LOCATION dynamorio                  # NON-PUBLIC
-    LOCATION${location_suffix})                                     # NON-PUBLIC
+  DynamoRIO_get_full_path(DR_TARGET_LOCATION dynamorio)             # NON-PUBLIC
```

Comment:
Update: yes, it passes.

Reply:
> They should be properties of the given target 

Hmm, so even though they seem like per-project(-dir) globals when we set them, they are stored per-target?

Reply:
This may be a "recent" cmake change: I'm pretty sure these were pure globals and there was no per-target value back in 2.x.  These days our min version is 3.2.  Are they per-target in 3.2 or is it more recent?

Reply:
Yes that was my understanding. For example, for LIBRARY_OUTPUT_DIRECTORY:

> This property is initialized by the value of the variable CMAKE_LIBRARY_OUTPUT_DIRECTORY if it is set when a target is created.

I think CMake initializes them at configure time. It's unclear to me when exactly, i.e. what if you change the CMAKE_ variables at configure time in a not-top-level CMakeLists.txt.

I have checked all occurrences though and the paths match before/after change. I am checking samples_proj in Windows 32-bit now.

Reply:
> -> nevermind, it's a 32-bit test

It is a 64-bit test as well: not limited to 32-bit.

Reply:
> Are they per-target in 3.2 or is it more recent?

Looks like yes, according to online docs.

Reply:
> It is a 64-bit test as well: not limited to 32-bit.

I misread the CMakeLists file. Nevertheless, it's passing in both 32 and 64.

I found that 2 tests are failing in 32-bit on master, unrelated to this PR. I'll file an issue or re-open and update i#2868.

Reply:
OK, I see why the test passes: these lines are not propagated to the public CMakeLists.txt.  I think if they were it would fail.

Reply:
You mean these lines stay within the DR project? Yes they do. But not following, which one is the 'public' CMakeLists.txt?